src/databases/databases.js

```diff
@@ -15,6 +24,18 @@ if (config.createDatabaseIfNotExist && !config.readOnly) {
   if (fs.existsSync(config.privateDBSchema)) privateDB.exec(fs.readFileSync(config.privateDBSchema).toString());
 }
 
+// Upgrade database if required
+if (!config.readOnly) {
+  let versionCode = db.prepare("SELECT code FROM version").get() || 0;
+  let path = config.schemaFolder + "/_upgrade_" + versionCode + ".sql";
+  while (fs.existsSync(path)) {
+    db.exec(fs.readFileSync(path).toString());
+
+    versionCode = db.prepare("SELECT code FROM version").get();
+    path = config.schemaFolder + "/_upgrade_" + versionCode + ".sql";
```

Comment:
I'd side on storing the version number that's already been executed rather than storing the version ahead - could bet confusing if people are not using the server but still using the update files.